src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -699,8 +699,10 @@ protected virtual ClusterTransactionBatchCollector CollectCommandsBatch(ClusterO
                         if (_logger.IsInfoEnabled)
                             _logger.Info($"Failed to execute cluster transaction batch (count: {batchCollector.Count}), will retry them one-by-one.", e);
 
-                        if (ExecuteClusterTransactionOneByOne(batch))
+                        if (ExecuteClusterTransactionOneByOne(batch, out var actualBatchSize, out var actualCommandsCount))
                             batchCollector.AllCommandsBeenProcessed = true;
+                        
+                        return (actualBatchSize, actualCommandsCount);
```

Comment:
We have to return here to prevent later code from trying to handle commands that were not executed.